languagetool-core/src/main/java/org/languagetool/JLanguageTool.java

```diff
@@ -1703,14 +1743,10 @@ public static void removeTemporaryFiles() {
             }
             RuleMatch newMatch = new RuleMatch(match);
             newMatch.setOffsetPosition(newFromPos, newToPos);
-            newMatch.setLine(range.from.line);
-            newMatch.setEndLine(range.to.line);
-            if (match.getLine() == 0) {
-              newMatch.setColumn(range.from.column + 1);
-            } else {
-              newMatch.setColumn(range.from.column);
-            }
-            newMatch.setEndColumn(range.to.column);
+            newMatch.setLine(from.line);
+            newMatch.setEndLine(to.line);
+            newMatch.setColumn(from.column - (from.line == 0 ? 1 : 0));
+            newMatch.setEndColumn(to.column - (to.line == 0 ? 1 : 0));
```

Comment:
Here's an inconsistency. Some clients expect column to be 0-based (as in javadoc) and some — 1-based. From the history I've seen that the similar condition in the previous version of code was a hack to make some tests happy (since they're using deprecated API anyway), so I've decided to create an equivalent.

Reply:
Line/column information is deprecated in LT, we don't show it anymore in the JSON response (our main use case), we just keep it running for the command-line LT.

Reply:
Any plans to remove it?

Reply:
I'd prefer to have it removed, but I see the line number is useful when running LT on the command-line, so it would need to be calculated somewhere else (maybe not in JLanguageTool). But I have no plans to work on this.

Reply:
Do you mean only `org.languagetool.commandline.Main` or anything else? If the former, I think this calculation could be moved there relatively easily, and `JLanguageTool` could become free of this logic, and maybe even faster.

Reply:
I haven't checked closely, but yes, `org.languagetool.commandline.Main` sounds like the logical place for this code.

Reply:
I can also see some column usages in XML serialization, Wikipedia and UI HTML. I don't know of their importance nowadays. But maybe I'll take a look at this some day :)